src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs

```diff
@@ -1,24 +1,54 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
 using System.Linq;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
 using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Model;
 
 namespace NuGet.Protocol
 {
     public class PackageSearchResourceV3 : PackageSearchResource
     {
+        private readonly HttpSource _client;
+        private readonly Uri[] _searchEndpoints;
+
+#pragma warning disable CS0618 // Type or member is obsolete
         private readonly RawSearchResourceV3 _rawSearchResource;
+#pragma warning restore CS0618 // Type or member is obsolete
 
+#pragma warning disable CS0618 // Type or member is obsolete
         public PackageSearchResourceV3(RawSearchResourceV3 searchResource)
+#pragma warning restore CS0618 // Type or member is obsolete
             : base()
         {
             _rawSearchResource = searchResource;
         }
 
+        public PackageSearchResourceV3(HttpSource client, IEnumerable<Uri> searchEndpoints)
+            : base()
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            if (searchEndpoints == null)
+            {
+                throw new ArgumentNullException("searchEndpoints");
```

Comment:
```suggestion
                throw new ArgumentNullException(nameof(searchEndpoints));
```

this should allow you to remove one of the new lines in the global suppression file.

Reply:
Done.